### PR TITLE
Add docs for entity routing.

### PR DESCRIPTION
**Pull request by markstory**
See cakephp/cakephp#11300

### CODE REVIEW

**Comment by saeideng** [comment]
can we use `virtual field `  into this?

**Comment by markstory** [comment]
@saeideng yes. Virtual fields are exposed as entity properties. Do you think that is worth calling out?

File: en/development/routing.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Routing
+oouting
```
Comment (okinaka) [inline en/development/routing.rst]: typo?

Comment (markstory) [reply en/development/routing.rst]: Yes very much so :blush:

File: en/development/routing.rst [hunk 2]
```diff
@@ -1319,6 +1319,55 @@ string URLs as the destination::
...
+
+    $routes->get(
+        '/view/:id',
+        ['controller' => 'Articles', 'action' => 'view'],
+        'articles:view'
+    );
+
+You can generate URLs to this route using::
+
+    Router::url(['_name' => 'articles:view', 'id' => $article->id]);
```
Comment (ravage84) [inline en/development/routing.rst]: Where is that ``$article`` coming from?

File: en/development/routing.rst [hunk 3]
```diff
@@ -1319,6 +1319,55 @@ string URLs as the destination::
...
+    // Create the route just like before.
+    $routes->get(
+        '/view/:id',
+        ['controller' => 'Articles', 'action' => 'view'],
+        'articles:view'
+    );
+
+Now we can generate URLs using the ``_entity`` key::
+
+    Router::url(['_name' => 'articles:view', '_entity' => $article]);
```
Comment (ravage84) [inline en/development/routing.rst]: Where is that ``$article`` coming from?

File: en/development/routing.rst [hunk 4]
```diff
@@ -1319,6 +1319,55 @@ string URLs as the destination::
...
+    );
+
+You can generate URLs to this route using::
+
+    Router::url(['_name' => 'articles:view', 'id' => $article->id]);
+
+Later on, you may want to expose the article slug in the URL for SEO purposes.
+In order to do this you would need to update everywhere you generate a URL to
+the ``articles:view`` route, which could take some time. If we use entity routes
+we pass the entire entity into URL generation allow us to skip any rework when
```
Comment (ravage84) [inline en/development/routing.rst]: the entire article entity

Comment (ravage84) [reply en/development/routing.rst]: allow -> allowing?

File: en/development/routing.rst (hunk 3 above)
Comment (ravage84) [reply en/development/routing.rst]: Just out of curiosity, does the entity route class  throw an exception if you do not provide ``_entity``?

**Comment by saeideng** [comment]
>@saeideng yes. Virtual fields are exposed as entity properties. Do you think that is worth calling out?

I asked  because  sometimes  we need to change filed/property name on DB/table/entity
is better to use Virtual fields for outputting  old filed name
by this way we no need to change all usage of entity  (include entity routing )

File: en/development/routing.rst (hunk 3 above)
Comment (markstory) [reply en/development/routing.rst]: Not if you include all the required parameters to create a route match.

**Pull request merged**